pkg/pluginmanager/manager.go

```diff
@@ -191,10 +191,12 @@ func DiscoverServerPlugins() ([]discovery.Discovered, error) {
 			discoveredPlugins[i].ContextName = context.Name
 
 			// Associate Target of the plugin based on the Context Type of the Context
-			switch context.Target {
-			case configtypes.TargetTMC:
+			switch context.ContextType {
+			case configtypes.ContextTypeTMC:
 				discoveredPlugins[i].Target = configtypes.TargetTMC
-			case configtypes.TargetK8s:
+			case configtypes.ContextTypeK8s:
+				discoveredPlugins[i].Target = configtypes.TargetK8s
+			case configtypes.ContextTypeTAE:
```

Comment:
I feel sharing knowledge through code reviews is valuable, so, although this is not a big deal, I'll just mention that you can put multiple values on the same `case` line.  

```
case configtypes.ContextTypeK8s, configtypes.ContextTypeTAE:
   discoveredPlugins[i].Target = configtypes.TargetK8s
```
You can test out this syntax here if you want: https://go.dev/play/p/Z2EjEjmrz3r

I find that is not only shorter but it would make this case more future-proof; looking at this code in a few months, someone may think using `configtypes.TargetK8s` for two different cases is a typo, and would waste a bit of time looking into it.  Putting it in the same case line makes it clear that it is on purpose.

But like I said, it is not a big deal.

Reply:
BUT, now that I think about it some more, it may be even more future-proof, to do:
```
switch context.ContextType {
   case configtypes.ContextTypeTMC:
      discoveredPlugins[i].Target = configtypes.TargetTMC
    default:
      // All other context types are associated with the kubernetes target
      discoveredPlugins[i].Target = configtypes.TargetK8s
```
This implies that TMC is the special case, and that we expect a future new context-type to be associated with a kubernetes target.  Then, when adding a new context-type, there is nothing to do.  Do you feel that is a reasonable assumption?

Reply:
> This implies that TMC is the special case, and that we expect a future new context-type to be associated with a kubernetes target. Then, when adding a new context-type, there is nothing to do. Do you feel that is a reasonable assumption?

I am not sure we should assume that any future context-type will be associated with `k8s` context-type. WDYT @vuil ?

Reply:
Actually thinking about this a bit more, I feel that keeping k8s as default makes sense. I will update the PR accordingly.